### PR TITLE
Switch pools

### PR DESCRIPTION
**Pull request by friedger**
This PR
* adds a test that shows how to switch pools without a cool down cycle

### CODE REVIEW

File: tests/integration/pox/stacking/pooled-stacking-switch-pools.spec.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        lockPeriodCycles: 1,
+      }
+    );
+    expect(response.error).toBeUndefined();
+
+    let [block, tx] = await waitForStacksTransaction(
+      orchestrator,
+      response.txid
+    );
+    expect(tx.success).toBeTruthy();
```
Comment (obycode) [inline tests/integration/pox/stacking/pooled-stacking-switch-pools.spec.ts:70]: Use `asyncExpectStacksTransactionSuccess` here?

File: tests/integration/pox/stacking/pooled-stacking-switch-pools.spec.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    // Check Pool operators/Cloe's table entry
+    const poxAddrInfo1 = await readRewardCyclePoxAddressListAtIndex(
+      network,
+      2,
+      0
+    );
+    expect(poxAddrInfo1?.["total-ustx"]).toEqual(uintCV(80_000_000_000_000));
+  });
+
+  it("user can swith pools and new pool operator can extend (cycle #1)", async () => {
```
Comment (obycode) [inline tests/integration/pox/stacking/pooled-stacking-switch-pools.spec.ts:103]: "switch"

File: tests/integration/pox/stacking/pooled-stacking-switch-pools.spec.ts [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    const network = new StacksTestnet({ url: orchestrator.getStacksNodeUrl() });
+
+    // Alice revokes delegates 90m STX to Bob
```
Comment (obycode) [inline tests/integration/pox/stacking/pooled-stacking-switch-pools.spec.ts:106]: "Alice revokes, then delegates"

File: tests/integration/pox/stacking/pooled-stacking-switch-pools.spec.ts [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    response = await broadcastStackAggregationCommitIndexed(
+      { poxVersion: 2, network, account: Accounts.WALLET_2, fee, nonce: 1 },
+      { poolRewardAccount: Accounts.WALLET_2, cycleId: 3 }
+    );
+    expect(response.error).toBeUndefined();
+    await asyncExpectStacksTransactionSuccess(orchestrator, response.txid);
+
+    let poxInfo = await getPoxInfo(network);
+
+    // Assert that the next cycle has 130m STX locked
```
Comment (obycode) [inline tests/integration/pox/stacking/pooled-stacking-switch-pools.spec.ts:151]: 80m

**Review by obycode — APPROVED** [review]
Looks good. Just minor comments.

**Comment by friedger** [comment]
I have updated the test to clarify that pox is disabled due to 2.2

It is not clear to me how I can delay epoch 2.2 by 1 cycle. If I change the timeline, the orchestrator does not start @obycode

**Comment by friedger** [comment]
When using a different block, would this result in failed start of the orchestrator?

**Comment by obycode** [comment]
I think there was some logic in the node to check the configuration and fail early if it was invalid, but I'd have to check. What settings are you using when you see this? Do you see the stacks-node start and quickly disappear, for example if you watch Docker Desktop?

**Comment by obycode** [comment]
Or is it running and hitting this error loop?

```
May  3 16:46:11.326477 INFO Anchor block selected for cycle 1: 7d14cb6651c71cfe872bf43c02c2805becdb7f84/4a643852e52434605e332fad93b74db5186a2fb4d002eb0266a7497b4a2ac568 (txid 58156f971ae5ff76010156eda0788538f4bc6311f91b559fbbde481f8f7cf44f)
May  3 16:46:11.326866 WARN Error processing new burn block: ChainstateError(ClarityError(Interpreter(Unchecked(NoSuchContract("ST000000000000000000002AMW42H.pox-2")))))
May  3 16:46:12.335212 INFO Anchor block selected for cycle 1: 7d14cb6651c71cfe872bf43c02c2805becdb7f84/4a643852e52434605e332fad93b74db5186a2fb4d002eb0266a7497b4a2ac568 (txid 58156f971ae5ff76010156eda0788538f4bc6311f91b559fbbde481f8f7cf44f)
May  3 16:46:12.336441 WARN Error processing new burn block: ChainstateError(ClarityError(Interpreter(Unchecked(NoSuchContract("ST000000000000000000002AMW42H.pox-2")))))
```

**Comment by friedger** [comment]
I tried different settings.
also
```
  epoch_2_2: Constants.DEVNET_DEFAULT_EPOCH_2_2 + 10,
  pox_2_unlock_height: Constants.DEVNET_DEFAULT_POX_2_UNLOCK_HEIGHT + 10,
  epoch_2_3: Constants.DEVNET_DEFAULT_EPOCH_2_3 + 10,
```
I don't see any logs from stacks-node, it looks like it never started or disappeared immediately

**Comment by friedger** [comment]
This one works:
```
    epoch_2_2: 128,
    pox_2_unlock_height: 129,
    epoch_2_3: 138,
```

**Comment by obycode** [comment]
> I don't see any logs from stacks-node, it looks like it never started or disappeared immediately

I've got an issue open in clarinet to add an option to keep the logs around.

**Pull request merged**